### ISSUE TITLE
Early detection of training not progressing

### ISSUE THREAD

**Issue by Ayush-Rawal** [NVIDIA/tacotron2#432]
Hi,

Thank you for your work and this excellent repository. As the model takes a long time to train, I wish to early detect if the model is training correctly or if I need to adjust some parameters.

I'm training the model for a Hindi language dataset using a warm start with the previously trained model on LJ Speech.
The dataset contains 2300 files of various lengths (most are 4-10s, with the highest being 20s) sampled at 48000 hz which I've downsampled to 22050. Silence has been added to the end of each file using https://github.com/Yeongtae/tacotron2/blob/master/preprocess_audio.py as recommended in #375 

1. I'm using the transliteration filters as unidecode was mostly able to transliterate transcriptions to how they'd sound in english as such I've also not made any changes to text/symbols (This is my biggest doubt, should I go with basic filters and adding language symbols to the list instead?)

2. My batch size is 8, as otherwise I get cuda memory errors. Is that fine or should I increase batch size while reducing other parameters?

3. I've currently run 22epochs (~6k steps). Is training going normally? Or is it too soon to tell?
Here are screenshots of my alignment image.

![3K Steps](https://user-images.githubusercontent.com/31813648/101243420-25f48980-3726-11eb-86ba-67cca8e1d62c.png)
![4K Steps](https://user-images.githubusercontent.com/31813648/101243421-27be4d00-3726-11eb-867d-332dc6358cf3.png)
![5K Steps](https://user-images.githubusercontent.com/31813648/101243422-2856e380-3726-11eb-8b8b-23aee16dbd8d.png)
![6K Steps](https://user-images.githubusercontent.com/31813648/101243423-28ef7a00-3726-11eb-8047-bb1762fcc882.png)

My hparams are:

```python
        ################################
        # Experiment Parameters        #
        ################################
        epochs=500,
        iters_per_checkpoint=1000,
        seed=1234,
        dynamic_loss_scaling=True,
        fp16_run=False,
        distributed_run=False,
        dist_backend="nccl",
        dist_url="tcp://localhost:54321",
        cudnn_enabled=True,
        cudnn_benchmark=False,
        ignore_layers=['embedding.weight'],

        ################################
        # Data Parameters             #
        ################################
        load_mel_from_disk=False,
        training_files='filelists/indictts_hindi_female_mono_train.txt',
        validation_files='filelists/indictts_hindi_female_mono_val.txt',
        text_cleaners=['transliteration_cleaners'],

        ################################
        # Audio Parameters             #
        ################################
        max_wav_value=32768.0,
        sampling_rate=24000,
        filter_length=1024,
        hop_length=256,
        win_length=1024,
        n_mel_channels=80,
        mel_fmin=0.0,
        mel_fmax=8000.0,

        ################################
        # Model Parameters             #
        ################################
        n_symbols=len(symbols),
        symbols_embedding_dim=512,

        # Encoder parameters
        encoder_kernel_size=5,
        encoder_n_convolutions=3,
        encoder_embedding_dim=512,

        # Decoder parameters
        n_frames_per_step=1,  # currently only 1 is supported
        decoder_rnn_dim=1024,
        prenet_dim=256,
        max_decoder_steps=1000,
        gate_threshold=0.5,
        p_attention_dropout=0.1,
        p_decoder_dropout=0.1,

        # Attention parameters
        attention_rnn_dim=1024,
        attention_dim=128,

        # Location Layer parameters
        attention_location_n_filters=32,
        attention_location_kernel_size=31,

        # Mel-post processing network parameters
        postnet_embedding_dim=512,
        postnet_kernel_size=5,
        postnet_n_convolutions=5,

        ################################
        # Optimization Hyperparameters #
        ################################
        use_saved_learning_rate=False,
        learning_rate=1e-3,
        weight_decay=1e-6,
        grad_clip_thresh=1.0,
        batch_size=8,
        mask_padding=True  # set model's padded outputs to padded values
```

Thanks

**Comment by Abhisitthqgam** [comment]
I think the model starting to overfitting may reduce the learning rate and add more data after training to see.(I have the same problem.)
